install.sh

```diff
@@ -491,6 +577,70 @@ build_from_tarball_icu()
     pop_directory
 }
 
+build_from_tarball_libpng()
+{
+    local URL=$1
+    local ARCHIVE=$2
+    local REPO=$3
+    local JOBS=$4
+    shift 4
+
+    if [[ !($BUILD_LIBPNG) ]]; then
+        initialize_libpng_packages
+        display_message "LIBPNG build not enabled"
+        return
+    fi
+
+    display_message "Download $ARCHIVE"
+
+    create_directory $REPO
+    push_directory $REPO
+
+    # Extract the source locally.
+    wget --output-document $ARCHIVE $URL
+    tar --extract --file $ARCHIVE --xz --strip-components=1
+
+    # Build and install.
+    configure_options $LIBPNG_LINK $LIBPNG_STANDARD ${prefix} "$@"
+    make_jobs $JOBS --silent
+    make install
+    configure_links
+
+    pop_directory
+}
+
+build_from_tarball_qrencode()
+{
+    local URL=$1
+    local ARCHIVE=$2
+    local REPO=$3
+    local JOBS=$4
+    shift 4
+
+    if [[ !($BUILD_QRENCODE) ]]; then
+        initialize_qrencode_packages
+        display_message "QRENCODE build not enabled"
+        return
+    fi
+
+    display_message "Download $ARCHIVE"
+
+    create_directory $REPO
+    push_directory $REPO
+
+    # Extract the source locally.
+    wget --output-document $ARCHIVE $URL
+    tar --extract --file $ARCHIVE --bzip2 --strip-components=1
+
+    # Build and install.
+    configure_options $QRENCODE_LINK $QRENCODE_STANDARD ${prefix} "$@"
+    make_jobs $JOBS --silent
+    make install
+    configure_links
+
+    pop_directory
+}
+
```

Comment:
We are violating DRY here. Now that we have identical tar builds we should generalize this function.